Sources/funnet/Core/NetworkResponder.swift

```diff
@@ -28,3 +18,31 @@ public struct NetworkResponder: NetworkResponderProtocol {
     
     public init() {}
 }
+
+public func stub<T: Codable>(_ responder: NetworkResponder, with model: T) -> (Fireable) -> Void {
+    return { _ in
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+            responder.dataHandler(try? JSONEncoder().encode(model))
+        }
+    }
+}
+
+public func stubNoDelay<T: Codable>(_ responder: NetworkResponder, with model: T) -> (Fireable) -> Void {
+    return { _ in
+        responder.dataHandler(try? JSONEncoder().encode(model))
+    }
+}
+
+public func stubHTTPResponse<T: NetworkCall>(with statusCode: Int) -> (T) -> Void {
+    return { call in
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+            call.responder.httpResponseHandler(HTTPURLResponse(url: call.baseUrl.url!, statusCode: statusCode, httpVersion: nil, headerFields: nil)!)
+        }
+    }
+}
+
+public func stubHTTPResponseNoDelay<T: NetworkCall>(with statusCode: Int) -> (T) -> Void {
```

Comment:
One thing that pained me in the early stages of development was that sometimes a viewmodel would depend on both the response's data and the status code, and we didn't have a good way to stub both of them because we were manually setting the firingFunc. I like this a lot more.